buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes

```diff
@@ -0,0 +1,238 @@
+                    }
+
+                    def nodeLabels = []
+                    if (aNode.getLabelString()) {
+                        nodeLabels.addAll(aNode.getLabelString().tokenize(' '))
+                    }
+
+                    // cache job
+                    jobs["${nodeName}"] = {
+                        node("${nodeName}") {
+                            if (MODES.contains('cleanup')) {
+                                stage("${nodeName} - Cleanup Workspaces") {
+                                    def buildWorkspace = "${env.WORKSPACE}"
+                                    if (nodeLabels.contains('sw.os.windows')) {
+                                        // convert windows path to unix path
+                                        buildWorkspace = sh(script: "cygpath -u '${env.WORKSPACE}'", returnStdout: true).trim()
+                                    }
+
+                                    // Cleanup Test results
+                                    sh "cd /tmp && pwd && rm -rf ${cleanDirs.toString().replaceAll(',',' ')[1..-1]}"
+
+                                    if (nodeLabels.contains('sw.os.windows')) {
+                                        sh """
+                                            cd ${buildWorkspace}/../../ && pwd && rm -rf ${cleanDirs.toString().replaceAll(',',' ')[1..-1]}
+                                            # Remove shared classes cache on Windows
+                                            rm -rf /tmp/javasharedresources /temp/javasharedresources ${buildWorkspace}/../../javasharedresources
+                                        """
+                                    }
+
+                                    // Cleanup OSX shared memory
+                                    if (nodeLabels.contains('sw.os.osx')) {
+                                        retry(2) {
+                                            sh """
+                                                ipcs -ma
+                                                ipcs -ma | awk '/^m / { if (\$9 == 0) { print \$2 }}' | xargs -n 1 ipcrm -m
+                                                ipcs -ma
+                                            """
+                                        }
+                                    }
+
+                                    // Clean up defunct pipelines workspaces
+                                    dir("${buildWorkspace}/../") {
+                                        retry(3) {
+                                            if (nodeLabels.contains('sw.os.aix')) {
+                                                sh "rm -rf Build* Test* PullRequest*"
+                                                sh "ls -al"
+                                            } else {
+                                                sh "ls | grep -v ${JOB_NAME} | xargs rm -rf"
+                                            }
+                                            sh "sleep 1m"
```

Comment:
Can this `sleep` move to the start of the 'retry' and put inside an `if` so that we only sleep if we are doing a `retry`?

Reply:
The `sleep` should be in between `retry` only if the block failed.

Reply:
see
https://issues.jenkins-ci.org/browse/JENKINS-54735?focusedCommentId=363543&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363543

Reply:
It's been already fixed, see L176.